python/cugraph/cugraph/gnn/comms/cugraph_nccl_comms.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+from raft_dask.common.nccl import nccl
+from raft_dask.common.comms_utils import inject_comms_on_handle_coll_only
+
+from pylibraft.common.handle import Handle
+from rmm._cuda.gpu import getDevice, setDevice
+
+from cugraph.dask.comms.comms_wrapper import init_subcomms
+
+__nccl_comms = None
+__raft_handle = None
+
+
+def nccl_init(rank: int, world_size: int, uid: int):
+    try:
+        ni = nccl()
+        ni.init(world_size, uid, rank)
+        return ni
+    except Exception as ex:
+        raise RuntimeError(f"A nccl error occurred: {ex}")
+
+
+def make_raft_handle(
+    rank, world_size, nccl_comms, n_streams_per_handle=0, verbose=False
+):
+    handle = Handle(n_streams=n_streams_per_handle)
+    inject_comms_on_handle_coll_only(handle, nccl_comms, world_size, rank, verbose)
+
+    return handle
+
+
+def __get_2D_div(ngpus):
+    prows = int(math.sqrt(ngpus))
+    while ngpus % prows != 0:
+        prows = prows - 1
+    return prows, int(ngpus / prows)
```

Comment:
Just FYI, maybe in the future, we may use a common C++ utility function to compute a desirable 2D partition (with an user option to override the default). Setting prows as close as possible to math.sqrt(ngpus) is just one possibility.

Reply:
I copied this from the existing dask config.  I would be happy to replace this with a better function if that becomes available.  We can make an issue for it.